### ISSUE TITLE
[EPIC] Facility Locator: AutoSuggest for "Service type"

### ISSUE THREAD

**Issue by davidmpickett** [department-of-veterans-affairs/va.gov-cms#15541]
## Background

The Service Type field is one of the three primary fields on Facility Locator. Depending on the selection for Facility type, Service type may be:
- **Required** (Community providers)
- **Optional** (VA health, Urgent care, Emergency care, VA benefits)
- **Not available** (Community pharmacies, VA cemeteries, Vet Centers)

Currently the Service Type field uses two different interaction patterns:
- **AutoSuggest** (Community providers)
- **Drop down menu** (VA health, Urgent care, Emergency care, VA benefits)

Once Facility Locator [moves to Facilities API v1](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14507), the number of services for VA health, VA benefits, & Vet Centers will increase substantially and a drop down menu will no longer be viable. These will have to transition to AutoSuggest.

Additionally, we may be able to leverage the VA Service taxonomy to surface results based on plain-language names, so that Veterans don't need to know the exact technical term.

### Research
Requires usability study in Staging before shipping to prod

## User story
As a Veteran, I want to be able to search for a VA service provided at my closest VA location so that I can choose the facility which meets my needs. 
As a Veteran, I want the search results to accurately reflect the services available at the facilities returned. 

## Background
In Phase 1, we updated Vets-API to point to Lighthouse v1 to unblock teams dependent on this functionality (#15344). 
In Phase 2, we converted the Facility Locator to use Facility API v1 with existing functionality (#12907).
This work will update Facility Locator search by service for all of the ~90 services we can receive from Lighthouse v1. 

### Current behavior
The current search experience allows Veterans to search for 19 VA health services by selecting "All" or a specific service from a dropdown. The Facility Locator then queries Lighthouse to get the facilities associated with the specified service as defined by a legacy data source. This data source has not been entirely accurate for some time and is no longer maintained given the launch of Lighthouse Facilities V1 in August 2023. 
<details>
<summary> VA health search screenshot for reference </summary>

![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/878d18db-16cd-49bd-b246-8067d1eadd45)
</details>

### Desired behavior
We can now implement Lighthouse v1 and provide a list of possible VA health, benefit, and Vet Center services which is complete, accurate, and can leverage the taxonomy data for plain language and other keywords. 

MVP for this effort is VA health services  (+ Vet Centers, as a stretch)

**For example:** 
- **VA service taxonomy:** Optometry 
- **VAMC Patient-friendly Name:** Vision care, corrective lenses and eyeglasses (currently formatted as a comma-separated string, ticket submitted to change this to an array, consider additional work for front end depending on either of these formats)
- **VAMC Description:** Our optometrists offer you routine eye exams, preventive vision testing and treatment for conditions like glaucoma. We also provide prescriptions for eyeglasses and other assistive devices. 
- **VAMC Common Conditions:** vision exams, prescription eyeglasses, contact lenses (similar to patient-friendly name)

Veterans can enter keywords such as "optometry", "vision", "eyeglasses", "eye exam", "glaucoma", and "eye" to search for facilities with Optometry services.  

[Figma designs](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=0-1)

### Approach
SPIKE results: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15054#issuecomment-1743766937

The plan for this search has been to 
1. publish a static Json from CMS which contains the fields/details/keywords from the taxonomy which is made available on page load and refreshed during content publishing cycle
2. based on the words entered/selected by the Veteran, a parameter ID is used to query Lighthouse for facilities with that requested service

## ACs (draft)
- [ ] When Facility type = VA health, a Veteran can enter search parameters using plain language terms
- [ ] Search functionality for all other types of facilities is maintained
- [ ] Veterans (including those using screen readers, screen magnification, or keyboard navigation)
  - [ ] are aware of the type of search used by the field
  - [ ] are aware of the option to search for all VA health services
  - [ ] are aware of the option to search using a popular search suggestion
  - [ ] can select the option to search for all VA health services or one of the popular service suggestions
  - [ ] are aware of services which match the Veteran's search term
  - [ ] can select a matching service to search
  - [ ] are presented with an appropriate error message if the search terms do not match keywords in the service taxonomy
- [ ] When reached by keyboard navigation, the list of popular service suggestions and matching search terms receives   focus
- [ ] Search results are presented with a header which includes the search parameters and other existing behavior
- [ ] Search parameters remain visible after search results are displayed
- [ ] ( **may require design**) Error messages / intended behavior are verified for 
  - [ ] situations in which the search terms do not match keywords in the service taxonomy --
  - [ ] situations in which no results are found for the specified service and location -- examples: only 12 locations offer Bariatric surgery, 3 locations offer Genomic medicine, and 6 offer Hospital medicine
- [ ] Review and sign-off by Design, A11y, PM, and PO

**Comment by jilladams** [comment]
@Agile6MSkinner I stubbed: 
1. [Create QA test plan for Services autosuggest #20285](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20285)
2. [Run QA test plan for Services autosuggest #20286](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20286)

They're empty so far.

**Comment by mmiddaugh** [comment]
**Update 2.14.2025**

- Engineering nearly complete
- Research planning in progress

**Comment by mmiddaugh** [comment]
Leaving a trail for myself and others: I started the conversation with VHA DM Taxonomy SMEs (Steve Tokar, Jeff Grandon, and Justin Warren) using [this Sharepoint document](https://dvagov-my.sharepoint.com/:x:/r/personal/suzanne_middaugh_va_gov/Documents/va-services_for%20typeahead%20(version%201).xlsx?d=w83091d2beaa84612a64fbc6a9f4d42ea&csf=1&web=1&e=WnXVHE) to fill in keyword gaps

**Comment by mmiddaugh** [comment]
Closing this in favor of #20276
